### DISCUSSION TITLE
Unable to update app with NSOSStatusErrorDomain code -67048

### DISCUSSION THREAD

**Discussion by tzeejay** [sparkle-project/Sparkle#2299] [Q&A] [answered]
Hello everybody,

I have been searching for this for a week now and have tried all kinds of things to resolve this problem without any luck. I hope this is the right place to ask these questions.

We have shipped our last update in February. I wanted to ship an update last week but ended up with a Sparkle update UI that would never complete and really strange logs to which I am unable to find a sensible answer to. Previous updates had been successful without any problems. The only code signing differences between the last update and now is that Apple invalidated our provisioning profiles somehow. I created new ones without any problems. On top of all of this is that this update installs without any problems if manually updated by dragging the app into ´/Applications/`

The main problem log entry that this emits appears to be this line
`Error registering stapled ticket: Error Domain=NSOSStatusErrorDomain Code=-67048 "(null)"`

The app is code signed and notarized properly as far as I can tell. If anybody has seen this before or has any ideas I am all ears as I have no idea what to do at this point. This particular screenshot is a from a VM but this behaves exactly the same on a real Mac on both Intel or Apple Silicon.

<img width="1472" alt="Bildschirmfoto 2022-12-07 um 21 02 29" src="https://user-images.githubusercontent.com/1497755/207131977-1a01a355-e7f2-47c1-a189-e7b76facf8bc.png">

**Comment by zorgiepoo** [comment]
The last error that is cut off with a backtrace looks concerning. What is it?

Check any errors/logs from Autoupdate and Updater processes too. 

A screenshot of console logs is not that helpful compared to a dump of the logs. I think maybe log collect (or show) —last 5m would do that. 

Also what version of Sparkle is being used?

**Reply by tzeejay** [reply]
The version of our app that I am trying to update has been using Sparkle `1.27.1`. I have tried to update it with this version and with 2.3 and neither work.

I am trying now to get you all the other information

**Reply by tzeejay** [reply]
@zorgiepoo Thank you for your reply. 
This is from the same VM. I tried the upgrade again with it as well as with another Mac and observed no change, maybe something stands out to any of you

[update-log.txt](https://github.com/sparkle-project/Sparkle/files/10218078/update-log.txt)

**Comment by zorgiepoo** [comment, marked as answer]
```
default	13:52:07.194122+0100	Guardian	Warning: could not remove quarantine metadata from /Users/cj/Library/Containers/com.sudosecuritygroup.Guardian/Data/Library/Caches/com.sudosecuritygroup.Guardian/org.sparkle-project.Sparkle/Guardian (Autoupdate).app: Error Domain=NSPOSIXErrorDomain Code=1 "Operation not permitted" UserInfo={NSLocalizedDescription=Failed to remove file quarantine on Guardian (Autoupdate).app.}. This may not be a problem, the update process will continue.
error	13:52:07.203602+0100	Guardian	launch path not accessible
error	13:52:07.204901+0100	Guardian	(
	0   CoreFoundation                      0x00000001a3d41148 __exceptionPreprocess + 240
	1   libobjc.A.dylib                     0x00000001a3a8be04 objc_exception_throw + 60
	2   Foundation                          0x00000001a4c26d0c -[NSConcreteTask launchWithDictionary:error:] + 4608
	3   Foundation                          0x00000001a4c4f4dc +[NSTask launchedTaskWithLaunchPath:arguments:] + 160
	4   Sparkle                             0x00000001026f2a18 -[SUBasicUpdateDriver installWithToolAndRelaunch:displayingUserInterface:] + 4432
	5   Sparkle                             0x000000010270ac90 -[SUUIBasedUpdateDriver installWithToolAndRelaunch:] + 352
	6   AppKit                              0x00000001a6a2a5c8 -[NSApplication(NSResponder) sendAction:to:from:] + 460
	7   AppKit                              0x00000001a6a2a3c4 -[NSControl sendAction:to:] + 96
	8   AppKit                              0x00000001a6a2a2cc __26-[NSCell _sendActionFrom:]_block_invoke + 152
	9   AppKit                              0x00000001a6a2a1c0 -[NSCell _sendActionFrom:] + 204
	10  AppKit                              0x00000001a6a2a0e4 -[NSButtonCell _sendActionFrom:] + 104
	11  AppKit                              0x00000001a6a26f30 NSControlTrackMouse + 1752
	12  AppKit                              0x00000001a6a2682c -[NSCell trackMouse:inRect:ofView:untilMouseUp:] + 160
	13  AppKit                              0x00000001a6a266a0 -[NSButtonCell trackMouse:inRect:ofView:untilMouseUp:] + 740
	14  AppKit                              0x00000001a6a25910 -[NSControl mouseDown:] + 632
	15  AppKit                              0x00000001a6a23d80 -[NSWindow(NSEventRouting) _handleMouseDownEvent:isDelayedEvent:] + 4528
	16  AppKit                              0x00000001a6997100 -[NSWindow(NSEventRouting) _reallySendEvent:isDelayedEvent:] + 2444
	17  AppKit                              0x00000001a699650c -[NSWindow(NSEventRouting) sendEvent:] + 348
	18  AppKit                              0x00000001a69954b4 -[NSApplication(NSEvent) sendEvent:] + 2780
	19  AppKit                              0x00000001a6c4f0f4 -[NSApplication _handleEvent:] + 76
	20  AppKit                              0x00000001a6817008 -[NSApplication run] + 636
	21  AppKit                              0x00000001a67e86fc NSApplicationMain + 1132
	22  dyld                                0x000000010263508c start + 520
```

Sparkle 1.x does not support sandboxing. Sparkle 2 has an guide on the website for enabling its XPC Services for sandboxed apps

**Reply by tzeejay** [reply]
@zorgiepoo yeah I noticed that as well. Our app had been working like this before though so I am very confused. I am going to give this another try with the sandboxed Sparkle 2 version.

Does the installed app that is being updated include Sparkle 2 for that or will this work if Sparkle 2 is contained within the new update? I am not quite sure which process is running when in this entire process

**Reply by zorgiepoo** [reply]
The old app that is running the update process needs a correct configuration of Sparkle which is either Sparkle 1 without having the app sandboxed or Sparkle 2 following the website guide for enabling the XPC services and applying process entitlements. Be sure to test the update process. For any old version that “used to work” has a real underlying reason for it.

**Comment by tzeejay** [comment]
@zorgiepoo I just sat down and updated Sparkle + the entitlements to support sandboxing since we do want to be in the MAS at some point once Apple fixes the provisioning profile bugs. Worked first time with zero problems 🎉 

Thank you very much for your help here! I had completely forgotten that I had enabled sandboxing late last year while trying to get us into the MAS. I am so happy that this is now resolved
